src/plugins/es_ui_shared/static/forms/hook_form_lib/hooks/use_field_from_props.ts

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import { useMemo, useEffect } from 'react';
+import { UseFieldProps } from '../components';
+import { FieldHook, FieldConfig, FieldValidationData } from '../types';
+import { useFormContext } from '../form_context';
+import { useField, InternalFieldConfig } from './use_field';
+
+/**
+ * Initially the code in this hook was inside the <UseField /> component.
+ * It serves to initialize a FieldHook (useField() from "hooks/use_field.ts") based on the props
+ * passsed to the component.
+ * When refactoring the <UseMultiFields /> component I realized that it would be much simpler
+ * to extract the code into a separate hook and consume it from there too.
+ *
+ * @param props The props passed to <UseField /> or <UseMultiFields />
+ * @returns The field hook and props to forward to component to render for the field
+ */
+
+export const useFieldFromProps = <T, FormType, I>(
+  props: UseFieldProps<T, FormType, I>
+): { field: FieldHook<T, I>; propsToForward: { [x: string]: unknown } } => {
+  const form = useFormContext<FormType>();
+  const { getFieldDefaultValue, __readFieldConfigFromSchema, __updateDefaultValueAt } = form;
+
+  const {
+    path,
+    config = __readFieldConfigFromSchema<T, FormType, I>(props.path),
+    defaultValue,
+    component,
+    componentProps,
+    readDefaultValueOnForm = true,
+    onChange,
+    onError,
+    children,
+    validationData,
+    validationDataProvider,
+    ...rest
+  } = props;
+
+  const initialValue = useMemo<T>(() => {
+    // The initial value of the field.
+    // Order in which we'll determine this value:
```

Comment:
Is there a reason why we allow passing default value in so many different ways?

Reply:
Yes for flexibility for the consumer. All of those way have been used in one way or another in the forms, depending on the need and how the form is composed.

The "usual" way is to declare the defaultValue on the schema (correspond to an empty field initial state when there is no value passed) and the defaultValue on the form (field initial value).